src/librustc_mir/build/block.rs

```diff
@@ -79,4 +80,31 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             block.unit()
         })
     }
+
+    // Helper method for generating MIR inside a conditional block.
+    pub fn with_cond<F>(&mut self, block: BasicBlock, span: Span,
```

Comment:
I was just emulating a pattern from trans, but returning a pair of blocks makes sense